pkg/webhook/admission/validator.go

```diff
@@ -0,0 +1,169 @@
+	NicPluggedID = CheckID("nic.plugged")
+	// NicVNicPortMirroringID defines an ID of a vnic_profile.port_mirroring == true check
+	NicVNicPortMirroringID = CheckID("nic.vnic_profile.port_mirroring")
+	// NicVNicPassThroughID defines an ID of a vnic_profile.pass_through == 'enabled' check
+	NicVNicPassThroughID = CheckID("nic.vnic_profile.pass_through")
+	// NicVNicCustomPropertiesID defines an ID of a vnic_profile.custom_properties presence check
+	NicVNicCustomPropertiesID = CheckID("nic.vnic_profile.custom_properties")
+	// NicVNicNetworkFilterID defines an ID of a vnic_profile.networ_filter presence check
+	NicVNicNetworkFilterID = CheckID("nic.vnic_profile.network_filter")
+	// NicVNicQosID defines an ID of a vnic_profile.qos presence check
+	NicVNicQosID = CheckID("nic.vnic_profile.qos")
+)
+
+var validInterfaceDeviceModels = map[string]*struct{}{"e1000": nil, "rtl8139": nil, "virtio": nil}
+
+// CheckID identifies validation check for Virtual Machine Import
+type CheckID string
+
+// ValidationFailure describes Virtual Machine Import validation failure
+type ValidationFailure struct {
+	// Check ID
+	ID CheckID
+	// Verbose explanation of the failure
+	Message string
+}
+
+func validateNics(nics []*ovirtsdk.Nic) []ValidationFailure {
+	var failures []ValidationFailure
+	for _, nic := range nics {
+		failures = append(failures, validateNic(nic)...)
+	}
+	return failures
+}
+
+func validateNic(nic *ovirtsdk.Nic) []ValidationFailure {
+	var results []ValidationFailure
+	var nicID = ""
+	if id, ok := nic.Id(); ok {
+		nicID = id
+	}
+
+	//TODO: Networking rule #1
+
+	if failure, valid := isValidNicInterfaceModel(nic, nicID); !valid {
+		results = append(results, failure)
+	}
+	if failure, valid := isValidNicOnBoot(nic, nicID); !valid {
+		results = append(results, failure)
+	}
+	if failure, valid := isValidNicPlugged(nic, nicID); !valid {
+		results = append(results, failure)
+	}
+
+	if vNicProfile, ok := nic.VnicProfile(); ok {
+		if failure, valid := isValidVnicPortMirroring(vNicProfile, nicID); !valid {
+			results = append(results, failure)
+		}
+		if failure, valid := isValidVnicProfilePassThrough(vNicProfile, nicID); !valid {
+			results = append(results, failure)
+		}
+		if failure, valid := isValidVnicProfileCustomProperties(vNicProfile, nicID); !valid {
+			results = append(results, failure)
+		}
+		if failure, valid := isValidVnicProfileNetworFilter(vNicProfile, nicID); !valid {
+			results = append(results, failure)
+		}
+		if failure, valid := isValidVnicProfileQos(vNicProfile, nicID); !valid {
+			results = append(results, failure)
+		}
+	}
+
+	return results
+}
+
+func isValidNicInterfaceModel(nic *ovirtsdk.Nic, nicID string) (ValidationFailure, bool) {
+	iFace, ok := nic.Interface()
+	if _, found := validInterfaceDeviceModels[string(iFace)]; !ok || !found {
+		return ValidationFailure{
+			ID:      NicInterfaceCheckID,
+			Message: fmt.Sprintf("interface %s uses model %s that is not supported.", nicID, iFace),
```

Comment:
worth to list here the supported models to let the user know to which type the nic could be changed.

Reply:
Will do.